src/aspire/denoise/covar2d.py

```diff
@@ -33,12 +45,60 @@ def get_covar(self, coeffs=None, mean_coeff=None,  do_refl=false):
         :return: The covariance matrix of coefficients for all images.
 
         """
-        pass
+        if coeffs is None:
+            raise RuntimeError('The coefficients need to be calculated!')
+        if mean_coeff is None:
+            mean_coeff = self.get_mean(coeffs)
+
+        covar_coeff = []
+        ind = 0
+        ell = 0
+        mask =  self.basis._indices["ells"] == ell
+        coeff_ell = coeffs[mask, ...] - mean_coeff[mask, ...]
+        # covar_ell = np.zeros((coeff_ell.shape[0], coeff_ell.shape[0]))
+        covar_ell = np.array(coeff_ell @ coeff_ell.T/np.size(coeffs, 1))
+        #print("covar_ell", np.size(covar_ell))
+        covar_coeff.append(covar_ell)
+        ind += 1
```

Comment:
Why was `ind = 0` and not `1` to begin with?

Reply:
The ind = 0 case was used before the for loop.